### ISSUE TITLE
SSL (https) breaks flex queries in Internet Explorer with Error #2032: Stream Error

### ISSUE THREAD

**Issue by kentbower** [kbower/tickettest1#123]
_Imported from trac ticket 123.
Created by **kbower**
Opened in trac: Jan 07, 2010, 9:12:45 AM
Last modified in trac: Jan 28, 2010, 8:06:25 AM_
***
_by kbower:_
This problem is due to IE browser caching.  With no-cache HTTP header, internet explorer deletes the data before it can hand it off to the active-x flash player. See [this blog](http://www.robsondesign.com/blog/index.php/2008/08/30/ie-ssl-xml-https-swf-doa/).

**Comment by kentbower** [comment]
_**kbower** trac comment on Jan 07, 2010, 9:16:53 AM:_

(In [[155]](../commit/b5633f7a628d9b530b7fd33298712c060ddaa8d2)) This is a work-around, temporary fix for IE and https.  See #123

**Comment by kentbower** [comment]
_**kbower** trac comment on Jan 07, 2010, 9:21:57 AM:_

We think the long term solution is to extend the expose decorator to change the headers as needed.

```

from tg import response

def my_expose(func):
    """Decorator that sends headers that instruct browsers and proxies not to cache.
    """
    def newfunc(*args, **kwargs):
        response.headers['Cache-Control']='no-store'
        response.headers['Pragma']=''
        return func(*args, **kwargs)
    return newfunc
```

**Comment by kentbower** [comment]
_**kbower** trac comment on Jan 07, 2010, 9:24:07 AM:_

Above example based on [this web page](http://www.thesamet.com/blog/2006/07/14/making-ie-cache-less/).

**Comment by kentbower** [comment]
_**kbower** trac comment on Jan 19, 2010, 9:27:57 AM:_

(In [[170]](../commit/5dcd2707ff8628020605cece7707ec7c0838b6de)) Created new decorator, @expose_ie_json, which we use instead of @expose('json') in order to fix
SSL problem with Internet Explorer cache.  Fixes #123

**Comment by kentbower** [comment]
_**kbower** trac status change on Jan 19, 2010, 9:27:57 AM:_ **closed**

**Comment by kentbower** [comment]
_**kbower** trac comment on Jan 28, 2010, 8:06:25 AM:_

(In [[216]](../commit/9ac990e8e6ec4f6d3f949b39d5652f3729e9e0d1)) Fix yet another ie problem (see #123 ) where ie uses its cache, even though we've told it not to
